front/app/containers/Admin/projects/project/analysis/SelectedInputContext/index.tsx

```diff
@@ -10,6 +18,20 @@ const Context = createContext<{
 
 const SelectedInputContext = ({ children }) => {
   const [selectedInputId, setSelectedInputId] = useState<string | null>(null);
+  const [search] = useSearchParams();
+  const inputId = search.get('selected_input_id');
+
+  useEffect(() => {
```

Comment:
I think when an input is selected, the entry gets a grey background or so? I don't see which code is doing this though. Perhaps this happens at the component level or so?

Reply:
What happens here is that this context controls which input is selected (grey background in the list + it is shown in the preview column). When it detects `selected_input_id` in the search parameters, it updates the state of the analysis screen and removed the parameter from the url.